### PR TITLE
Fixes ads rewards are not refreshed after claiming an ad grant

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/10094

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

- See https://github.com/brave/brave-browser/issues/10094

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_rewards/browser/rewards_service_impl.h [hunk 1]
```diff
@@ -493,7 +493,8 @@ class RewardsServiceImpl : public RewardsService,
   void SetPublisherMinVisits(int visits) const override;
   void SetPublisherAllowNonVerified(bool allow) const override;
   void SetPublisherAllowVideos(bool allow) const override;
-  void UpdateAdsRewards() const override;
+  void UpdateAdsRewards(
+    const bool should_refresh) const override;
```
Comment (emerick) [inline components/brave_rewards/browser/rewards_service_impl.h:497]: Nit: Indentation seems off.

Comment (tmancey) [reply components/brave_rewards/browser/rewards_service_impl.h]: Fixed

File: browser/ui/webui/brave_rewards_page_ui.cc [hunk 2]
```diff
@@ -709,6 +709,10 @@ void RewardsDOMHandler::OnPromotionFinished(
       promotion.promotion_id,
       result,
       std::move(promotion_ptr));
+
+  if (promotion.type == 1) {  // ads
```
Comment (NejcZdovc) [inline browser/ui/webui/brave_rewards_page_ui.cc:713]: this should be done in `RewardsServiceImpl::OnAttestPromotion`

Comment (tmancey) [reply browser/ui/webui/brave_rewards_page_ui.cc]: Moved inside lib so same business logic is shared across all platforms

File: vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc [hunk 3]
```diff
@@ -872,12 +872,13 @@ ledger::ActivityInfoFilterPtr LedgerImpl::CreateActivityFilter(
                                                min_visits);
 }
 
-void LedgerImpl::UpdateAdsRewards() {
+void LedgerImpl::UpdateAdsRewards(
+  bool should_refresh) {
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc:876]: please move it to the same line

File: vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.h [hunk 4]
```diff
@@ -147,7 +147,8 @@ class LedgerImpl : public ledger::Ledger {
 
   void SetAutoContributeEnabled(bool enabled) override;
 
-  void UpdateAdsRewards() override;
+  void UpdateAdsRewards(
+      bool should_refresh) override;
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.h:151]: same as above

**Comment by tmancey** [comment]
CI failed on iOS, restarting iOS

**Pull request merged**